packages/samples/headless-react/src/components/instant-results/instant-results.fn.tsx

```diff
@@ -0,0 +1,92 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  SearchBox as HeadlessSearchBox,
+  InstantResults as HeadlessInstantResults,
+} from '@coveo/headless';
+
+interface InstantResultsProps {
+  controllerSearchbox: HeadlessSearchBox;
+  controllerInstantResults: HeadlessInstantResults;
+}
+
+export const InstantResults: FunctionComponent<InstantResultsProps> = (
+  props
+) => {
+  const {controllerSearchbox, controllerInstantResults} = props;
+  const isEnterKey = (e: React.KeyboardEvent<HTMLInputElement>) =>
+    e.key === 'Enter';
+  const [searchboxState, setStateSearchbox] = useState(
+    controllerSearchbox.state
+  );
+  const [instantResultsState, setStateInstantResults] = useState(
+    controllerInstantResults.state
+  );
+
+  useEffect(
+    () =>
+      controllerSearchbox.subscribe(() =>
+        setStateSearchbox(controllerSearchbox.state)
+      ),
+    []
+  );
+  useEffect(() =>
+    controllerInstantResults.subscribe(() =>
+      setStateInstantResults(controllerInstantResults.state)
+    )
+  );
+
+  return (
+    <div>
+      <p>
+        Type in the searchbox and hover a query suggestion to preview associated
+        results
+      </p>
+
+      <input
+        value={searchboxState.value}
+        onChange={(e) => controllerSearchbox.updateText(e.target.value)}
+        onKeyDown={(e) => isEnterKey(e) && controllerSearchbox.submit()}
+      />
+
+      <div style={{display: 'flex'}}>
+        <ul>
+          {searchboxState.suggestions.map((suggestion) => {
+            const value = suggestion.rawValue;
+            return (
+              <li
+                key={value}
+                onMouseEnter={() => controllerInstantResults.updateQuery(value)}
+                onClick={() => controllerSearchbox.selectSuggestion(value)}
```

Comment:
I would give a slightly more accessible example by moving those events into a `button` element inside the `li`.